docker/Dockerfile

```diff
@@ -90,9 +90,11 @@ RUN export SQLITE_AUTOCONF_ROOT=sqlite-autoconf-3430100 && \
     manylinux-entrypoint /build_scripts/build-sqlite3.sh
 
 COPY build_scripts/build-openssl.sh /build_scripts/
-RUN export OPENSSL_ROOT=openssl-1.1.1w && \
-    export OPENSSL_HASH=cf3098950cb4d853ad95c0841f1f9c6d3dc102dccfcacd521d93925208b76ac8 && \
+RUN export OPENSSL_ROOT=openssl-3.0.10 && \
+    export OPENSSL_HASH=1761d4f5b13a1028b9b6f3d4b8e17feb0cedc9370f6afe61d7193d2cdce83323 && \
     export OPENSSL_DOWNLOAD_URL=https://www.openssl.org/source && \
+    export OPENSSL_ROOT_LEGACY=openssl-1.1.1w && \
+    export OPENSSL_HASH_LEGACY=cf3098950cb4d853ad95c0841f1f9c6d3dc102dccfcacd521d93925208b76ac8 && \
```

Comment:
to be removed, I'd rather not build 2 openssl versions if we can avoid it.

Reply:
So, I think you don't need to open another PR, I can remove the existing SSL check, so the system's SSL version remains, and for Python, the downloaded version is used. I used a scheme similar to the Python doc: https://docs.python.org/3.10/using/unix.html#custom-openssl

Reply:
I'd rather be explicit than not, and once those EOL versions are removed, then just remove those blocks of code in Dockerfile.